dataflow/src/main/java/org/checkerframework/dataflow/cfg/node/ObjectCreationNode.java

```diff
@@ -44,32 +47,45 @@ public class ObjectCreationNode extends Node {
      *
      * @param tree the NewClassTree
      * @param enclosingExpr the enclosing expression Node if it exists, or null
-     * @param constructor the constructor node
+     * @param identifier the identifier node
      * @param arguments the passed arguments
      * @param classbody the ClassDeclarationNode
      */
     public ObjectCreationNode(
             NewClassTree tree,
             @Nullable Node enclosingExpr,
-            Node constructor,
+            Node identifier,
             List<Node> arguments,
             @Nullable ClassDeclarationNode classbody) {
         super(TreeUtils.typeOf(tree));
         this.tree = tree;
         this.enclosingExpression = enclosingExpr;
-        this.constructor = constructor;
+        this.identifier = identifier;
         this.arguments = arguments;
         this.classbody = classbody;
     }
 
     /**
-     * Returns the constructor node
+     * Returns the constructor node.
      *
      * @return the constructor node
+     * @deprecated use {@link #getIdentifier()}
      */
     @Pure
+    @Deprecated
     public Node getConstructor() {
-        return constructor;
+        return identifier;
+    }
+
+    /**
+     * Returns the identifier node. A non-generic constructor can refer to a ClassNameNode, while a
+     * generic constructor identifier can refer to a ParameterizedTypeNode.
```

Comment:
The comment about where annotations go is lost somewhere in the history. It might be worth documenting this, e.g. in the class javadoc, in case somebody wonders how to access annotations.